### DISCUSSION TITLE
Dreaded "maximum number of xxx certificates."

### DISCUSSION THREAD

**Discussion by sfurlani** [fastlane/fastlane#16872] [Q&A]
Hey,

I'm certain this is a user error and not a problem with Fastlane.

I'm using a single repo for multiple projects, specifying a branch for each (see `Matchfile` below).

I tried adding this new project/branch, and it gives me the dreaded "maximum # of certificates" error.

I really don't want to `nuke` everything and start all my projects over.
```
Traceback (most recent call last):
	20: from /usr/local/Cellar/fastlane/2.149.1/libexec/bin/fastlane:23:in `<main>'
	19: from /usr/local/Cellar/fastlane/2.149.1/libexec/bin/fastlane:23:in `load'
	18: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/bin/fastlane:23:in `<top (required)>'
	17: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/fastlane/lib/fastlane/cli_tools_distributor.rb:111:in `take_off'
	16: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/match/lib/match/commands_generator.rb:25:in `start'
	15: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/match/lib/match/commands_generator.rb:187:in `run'
	14: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	13: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	12: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	11: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	10: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	 9: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/match/lib/match/commands_generator.rb:53:in `block (2 levels) in run'
	 8: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/match/lib/match/runner.rb:93:in `run'
	 7: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/match/lib/match/runner.rb:160:in `fetch_certificate'
	 6: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/match/lib/match/generator.rb:33:in `generate_certificate'
	 5: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/cert/lib/cert/runner.rb:13:in `launch'
	 4: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/cert/lib/cert/runner.rb:44:in `run'
	 3: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/cert/lib/cert/runner.rb:176:in `create_certificate'
	 2: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/cert/lib/cert/runner.rb:181:in `rescue in create_certificate'
	 1: from /usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/fastlane_core/lib/fastlane_core/ui/ui.rb:17:in `method_missing'
/usr/local/Cellar/fastlane/2.149.1/libexec/gems/fastlane-2.152.0/fastlane_core/lib/fastlane_core/ui/interface.rb:141:in `user_error!': [!] Could not create another Development certificate, reached the maximum number of available Development certificates. (FastlaneCore::Interface::FastlaneError)
```

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.14.6                                                 |
| Ruby                        | 2.5.8                                                   |
| Bundler?                    | true                                                    |
| Git                         | git version 2.23.0                                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.149.1/libexec/bin/fastlane |
| Host                        | Mac OS X 10.14.6 (18G4032)                              |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.5/2.5.8/lib                    |
| OpenSSL Version             | OpenSSL 1.1.1f  31 Mar 2020                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 11.3                                                    |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./Fastfile`</summary>

```ruby
# This file contains the fastlane.tools configuration
# You can find the documentation at https://docs.fastlane.tools
#
# For a list of all available actions, check out
#
#     https://docs.fastlane.tools/actions
#
# For a list of all available plugins, check out
#
#     https://docs.fastlane.tools/plugins/available-plugins
#

# Uncomment the line if you want fastlane to automatically update itself
# update_fastlane

default_platform(:ios)

platform :ios do
  desc "Installs the certificates on the local machine"
  lane :certs do
	match(type: "development")
	# match "adhoc"
	# match appstore
  end
end

```
</details>

<details><summary>`./Appfile`</summary>

```ruby
app_identifier("xxxx") # The bundle identifier of your app
apple_id("ios_dev@xxxx.com") # Your Apple email address
team_id("xxxxx") # Your Apple team ID


# For more information about the Appfile, see:
#     https://docs.fastlane.tools/advanced/#appfile

```
</details>

<details><summary>`./Matchfile`</summary>

```ruby
git_url("git@github.com:xxxxxxxx/ios-certificates.git")

storage_mode("git")

type "development" # The default type, can be: appstore, adhoc, enterprise or development
  git_branch "xxxx_prototype"

# For all available options run `fastlane match --help`
# Remove the # in the beginning of the line to enable the other options

# The docs are available on https://docs.fastlane.tools/actions/match
```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.152.0 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| rake                      | 13.0.1       |
| CFPropertyList            | 3.0.2        |
| public_suffix             | 4.0.5        |
| addressable               | 2.7.0        |
| atomos                    | 0.1.3        |
| aws-eventstream           | 1.1.0        |
| aws-partitions            | 1.342.0      |
| aws-sigv4                 | 1.2.1        |
| jmespath                  | 1.4.0        |
| aws-sdk-core              | 3.104.0      |
| aws-sdk-kms               | 1.36.0       |
| aws-sdk-s3                | 1.74.0       |
| babosa                    | 1.0.3        |
| bundler                   | 2.0.2        |
| claide                    | 1.0.3        |
| colored                   | 1.2          |
| colored2                  | 3.1.2        |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| declarative               | 0.0.20       |
| declarative-option        | 0.1.0        |
| digest-crc                | 0.6.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| dotenv                    | 2.7.6        |
| emoji_regex               | 3.0.0        |
| excon                     | 0.75.0       |
| multipart-post            | 2.0.0        |
| faraday                   | 0.17.3       |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 0.14.0       |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| jwt                       | 2.2.1        |
| memoist                   | 0.16.2       |
| multi_json                | 1.15.0       |
| os                        | 1.1.0        |
| signet                    | 0.14.0       |
| googleauth                | 0.13.0       |
| httpclient                | 2.8.3        |
| mini_mime                 | 1.0.2        |
| uber                      | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.3        |
| google-cloud-errors       | 1.0.1        |
| google-cloud-core         | 1.5.0        |
| google-cloud-storage      | 1.26.2       |
| json                      | 2.3.1        |
| mini_magick               | 4.10.1       |
| plist                     | 3.5.0        |
| rubyzip                   | 2.3.0        |
| security                  | 0.1.3        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| slack-notifier            | 2.3.2        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| tty-screen                | 0.8.0        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| word_wrap                 | 1.0.0        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.17.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| xcpretty-travis-formatter | 1.0.0        |
</details>

*generated on:* **2020-07-16**
</details>

**Comment by randyvarela5** [comment]
you might need to login to developer.apple.com and delete a cert.  Thats usually how i get around this issue.

**Comment by joshdholtz** [comment]
@sfurlani Hey 👋 Multiple branches should only be used for multiple Apple developer teams. Apple only allows 2 distribution certs so you will hit your max on the third branch if you use multiple branches with the same team. Using 1 branch per team is ideal because you distribution cert can be used for all apps in that team 😊 Hope that helps!
